examples/mnist/run.py

```diff
@@ -57,51 +64,43 @@ def normalize(image, label):
   def _build_training_graph(self, training_data):
     """Build a graph for plaintext model training."""
 
-    # model parameters and initial values
-    w0 = tf.Variable(tf.random_normal([28 * 28, 512]))
-    b0 = tf.Variable(tf.zeros([512]))
-    w1 = tf.Variable(tf.random_normal([512, 10]))
-    b1 = tf.Variable(tf.zeros([10]))
-
+    model = keras.Sequential()
+    model.add(keras.layers.Flatten())
+    model.add(keras.layers.Dense(512, input_shape=[28, 28, 1]))
```

Comment:
Why do we have Flatten layer before a Dense layer with input_shape [28, 28, 1]?

Reply:
oups realized I had this in the notebook example... because `input_shape` is in the second layer, it gets ignored. that's why it can run even if the input shape after Flatten() is not [?, 28, 28, 1] but [?, 784]. I would recommend removing `input_shape=[28, 28, 1]` from the Dense layer or move it to the Flatten() layer.

Reply:
Or better yet, move the reshaping logic to the data pipeline function so that the data is already flattened by the time it gets here

Reply:
agreed, that would be better.